### DISCUSSION TITLE
What does `_metadata.bolt` do? Is it significant for the data stored or just used for caching?

### DISCUSSION THREAD

**Discussion by deepjyoti30** [zinclabs/zinc#589] [Q&A] [answered]
Hi there, as the question suggests, I would like to know the significance of the `_metadata.bolt` file. Wondering if this file is just used for caching purposes or is significant enough that removing it might lead to data loss.

**Comment by prabhatsharma** [comment, marked as answer]
> Hi there, as the question suggests, I would like to know the significance of the `_metadata.bolt` file. Wondering if this file is just used for caching purposes or is significant enough that removing it might lead to data loss.

Why do you want to delete it?

It stores metadata. e.g. credentials, index names, mappings, etc.. Deleting it will not lead to data loss theoretically but without metadats Zinc will not be able to provide you data, which means practically it will lead to data loss.

**Reply by siddharthlatest** [reply]
@prabhatsharma Follow-up question here:

- I've an upgrade zinc automation where on a new Zinc release, I upgrade my Zinc service. In this case, is `_metadata.bolt` safe to be kept as is, can a minor or major version in future intrroduce any changes that can prevent a server restart? (I've seen it being re-created by Zinc on a server re-start)
- One issue I've run into with keeping metadata.bolt is that credentials updated in a service file don't get reflected as they're likely cached in _metadata.bolt file. Only deleting this file allowed me to update the Basic Auth creds

**Reply by prabhatsharma** [reply]
metadata is not really recreated on every restart. credentials are provided through environment variables only on first start. After that environment variable loses its usefulness and credentials are stored in boltdb.

**Reply by siddharthlatest** [reply]
Thanks for the clarification!
